src/app/backend/replicasetlist.go

```diff
@@ -56,47 +58,132 @@ type ReplicaSet struct {
+	for _, replicaSet := range replicaSets {
 		var containerImages []string
-
 		for _, container := range replicaSet.Spec.Template.Spec.Containers {
 			containerImages = append(containerImages, container.Image)
 		}
 
+		matchingServices := getMatchingServices(services, &replicaSet)
+		var internalEndpoints []string
+		var externalEndpoints []string
+		for _, service := range matchingServices {
+			internalEndpoints = append(internalEndpoints,
+				getInternalEndpoint(service.Name, service.Namespace, service.Spec.Ports))
+			for _, externalIp := range service.Status.LoadBalancer.Ingress {
+				externalEndpoints = append(externalEndpoints,
+					getExternalEndpoint(externalIp.Hostname, service.Spec.Ports))
+			}
+		}
+
 		replicaSetList.ReplicaSets = append(replicaSetList.ReplicaSets, ReplicaSet{
-			Name:      replicaSet.ObjectMeta.Name,
-			Namespace: replicaSet.ObjectMeta.Namespace,
-			// TODO(bryk): This field contains test value. Implement it.
-			Description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. " +
-				"Nulla metus nibh, iaculis a consectetur vitae, imperdiet pellentesque turpis.",
-			Labels:          replicaSet.ObjectMeta.Labels,
-			PodsRunning:     replicaSet.Status.Replicas,
-			PodsPending:     replicaSet.Spec.Replicas - replicaSet.Status.Replicas,
-			ContainerImages: containerImages,
-			CreationTime:    replicaSet.ObjectMeta.CreationTimestamp,
-			// TODO(bryk): This field contains test value. Implement it.
-			InternalEndpoints: []string{"webapp"},
-			// TODO(bryk): This field contains test value. Implement it.
-			ExternalEndpoints: []string{"81.76.02.198:80"},
+			Name:              replicaSet.ObjectMeta.Name,
+			Namespace:         replicaSet.ObjectMeta.Namespace,
+			Description:       replicaSet.Annotations[DescriptionAnnotationKey],
+			Labels:            replicaSet.ObjectMeta.Labels,
+			PodsRunning:       replicaSet.Status.Replicas,
+			PodsPending:       replicaSet.Spec.Replicas - replicaSet.Status.Replicas,
+			ContainerImages:   containerImages,
+			CreationTime:      replicaSet.ObjectMeta.CreationTimestamp,
+			InternalEndpoints: internalEndpoints,
+			ExternalEndpoints: externalEndpoints,
 		})
 	}
 
-	return replicaSetList, nil
+	return replicaSetList
+}
+
+// Returns internal endpoint name for the given service properties.
+func getInternalEndpoint(serviceName string, namespace string, ports []api.ServicePort) string {
+	name := serviceName
+	if namespace != api.NamespaceDefault {
+		name = name + "." + namespace
+	}
+	var portsString []string
+	for _, port := range ports {
+		portsString = append(portsString, strconv.Itoa(port.Port)+"/"+string(port.Protocol))
+	}
+
+	if len(portsString) > 0 {
+		name = name + " " + strings.Join(portsString, ",")
+	}
+	return name
+}
+
+// Returns external endpoint name for the given service properties.
+func getExternalEndpoint(serviceIp string, ports []api.ServicePort) string {
+	name := serviceIp
+	var portsString []string
```

Comment:
How about extracting getPortsString() returning a string? It would be used twice.